examples/dmd/local_dw_csv.cpp

```diff
@@ -0,0 +1,889 @@
+                    else
+                    {
+                        dmd[curr_window] = new CAROM::NonuniformDMD(dim);
+                    }
+                    dmd[curr_window]->takeSample(sample, tval);
+                }
+            }
+
+            if (max_idx_snap == -1 && curr_window == numWindows-1)
+            {
+                curr_indicator_val = (indicator_idx.size() == 0) ? tval : sample[indicator_idx[numWindows]];
+                if (curr_indicator_val >= indicator_val[numWindows])
+                {
+                    max_idx_snap = idx_snap;
+                    indicator_val[numWindows] = curr_indicator_val;
+                    if (myid == 0)
+                    {
+                        cout << "State #" << idx_snap << " - " << data_filename 
+                             << " is the end of window " << numWindows-1 << "." << endl;
+                    }
+                    break;
+                }
+            }
+        }
+
+        if (indicator_idx.size() > 0)
+        {
+            numWindows = curr_window + 1;
+            dmd.resize(numWindows); 
+        }
+        CAROM_VERIFY(numWindows == curr_window + 1);
+
+        if (myid == 0)
+        {
+            cout << "Sampling on " << par_dir << " completed" << endl;
+            cout << "Samples loaded " << num_train_snap << endl;
+            cout << "Samples used " << max_idx_snap - min_idx_snap + 1 << endl;;
+            cout << "Number of windows: " << numWindows << endl;
+            if (windowNumSamples < infty)
+            {
+                csv_db.putDoubleVector(string(outputPath) + "/" + string(indicator_val_list) + ".csv", 
+                                       indicator_val, numWindows);
+            }
+        }
+
+        for (int window = 0; window < numWindows; ++window)
+        {
+            if (rdim != -1)
+            {
+                if (myid == 0)
+                {
+                    cout << "Creating DMD model #" << window << " with rdim: " << rdim << endl;
+                }
+                dmd[window]->train(rdim);
+            }
+            else if (ef != -1)
+            {
+                if (myid == 0)
+                {
+                    cout << "Creating DMD model #" << window << " with energy fraction: " << ef <<
+                         endl;
+                }
+                dmd[window]->train(ef);
+            }
+            dmd[window]->save(outputPath + "/window" + to_string(window));
+            if (myid == 0)
+            {
+                dmd[window]->summary(outputPath + "/window" + to_string(window));
+            }
+        }
+
+        dmd_training_timer.Stop();
+
+        if (ddt > 0.0)
+        {
+            CAROM::AdaptiveDMD* admd = nullptr;
+            CAROM::Vector interp_snap(dim, true);
+            vector<double> interp_error;
+
+            for (int window = 0; window < numWindows; ++window)
+            {
+                admd = dynamic_cast<CAROM::AdaptiveDMD*> (dmd[window]);
```

Comment:
I would recommend adding a quick check here to verify that the result of the dynamic_cast is not nullptr.

Reply:
A verififcation is added.